database/schema/ve_db_postgresql.sql

```diff
@@ -404,6 +405,7 @@ GRANT DELETE ON system_vulnerabilities, system_vulnerabilities_active, system_vu
 GRANT DELETE ON system_vulnerabilities, system_vulnerabilities_active, system_vulnerabilities_inactive TO ve_db_user_vmaas_sync;
 
 CREATE INDEX IF NOT EXISTS sv_rule_id ON system_vulnerabilities_active (rule_id);
+CREATE INDEX IF NOT EXISTS not_null_advisories_idx ON system_vulnerabilities_active (advisories) WHERE advisories IS NOT NULL;
```

Comment:
I think this index should be created for every partition separately as the query will be always for single account.

Reply:
Since I don't have experience with creating indexes I don't quite understand the request for creating index for every partition. I thought it is created for every partition, isn't it?
```
vulnerability=# select tablename, indexname, indexdef from pg_indexes where tablename in ('system_vulnerabilities_active_0', 'system_vulnerabilities_active_1', 'system_vulnerabilities_active_2') and indexdef like '%advisories%';
            tablename            |                   indexname                    |                                                                           indexdef
---------------------------------+------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------
 system_vulnerabilities_active_0 | system_vulnerabilities_active_0_advisories_idx | CREATE INDEX system_vulnerabilities_active_0_advisories_idx ON public.system_vulnerabilities_active_0 USING btree (advisories) WHERE (advisories IS NOT NULL)
 system_vulnerabilities_active_1 | system_vulnerabilities_active_1_advisories_idx | CREATE INDEX system_vulnerabilities_active_1_advisories_idx ON public.system_vulnerabilities_active_1 USING btree (advisories) WHERE (advisories IS NOT NULL)
 system_vulnerabilities_active_2 | system_vulnerabilities_active_2_advisories_idx | CREATE INDEX system_vulnerabilities_active_2_advisories_idx ON public.system_vulnerabilities_active_2 USING btree (advisories) WHERE (advisories IS NOT NULL)
(3 rows)
```
I can add `rh_account_id` to index if the query is going to be used for single account, is this what you mean?

Reply:
Sorry, this is my bad. I wrongly assumed there is a difference when index is created on partitioned parent table and when it's created on individual partitions (as it's done in the procedure few lines below). But [documentation](https://www.postgresql.org/docs/current/ddl-partitioning.html) says you are correct:

```
This automatically creates a matching index on each partition, and any partitions you create or attach later will also have such an index. An index or unique constraint declared on a partitioned table is “virtual” in the same way that the partitioned table is: the actual data is in child indexes on the individual partition tables.
```